### ISSUE TITLE
BUG [12.0] [fiscal_epos_print] arrotondamento sbagliato nella comunicazione dei dati alla stampante

### ISSUE THREAD

**Issue by matteoopenf** [OCA/l10n-italy#3721]
## Module
fiscal_epos_print

## Describe the bug
con un cliente abbiamo rilevato che in alcuni casi escono degli scontrini sbagliati.
Ossia Odoo a video mostra uno scontrino, ma sulla stampante epson ne esce un altro.
Dopo svariati controlli abbiamo concluso che la visione a video e' esatta ed e' lo scontrino cartaceo ad essere errato.
Qello che accade e' che in alcuni casi l'arrotondamento lato javascript e' errato

## To Reproduce
**Affected versions**:

- [ ] 12.0
- [ ] 14.0 ancora nessun cliente mi ha segnalato scontrini errati

1. creare uno scontrino con un prodotto che costa 16,36 con iva al 10%
2. stampare lo scontrino 
3. si otterra' a video 18 euro di scontrino e cartaceo 17,99

**Expected behavior**
entrambi gli scontrini devono combaciare 

**Additional context**
Add any other context about the problem here. (e.g. OS, Python version, ...)

**Comment by matteoopenf** [comment]
abbiamo elaborato un workround e a breve apro la pr

**Comment by pezcl67** [comment]
Ciao,
ho proprio questo tipo di problema con la epson.
La modifica fatta è testabile? Verrà messa nel branch a breve?

Grazie mille

**Comment by matteoopenf** [comment]
Si si runboat, prova pure 
Comunque da alcune settimane è in produzione dal cliente in cui avevamo il problema

**Comment by matteoopenf** [comment]
> Ciao, ho proprio questo tipo di problema con la epson. La modifica fatta è testabile? Verrà messa nel branch a breve?
> 
> Grazie mille

ho sistemaro runboat testala pure e se ok approva cosi c'e' possibilita' di merge

**Comment by pezcl67** [comment]
> > Ciao, ho proprio questo tipo di problema con la epson. La modifica fatta è testabile? Verrà messa nel branch a breve?
> > Grazie mille
> 
> ho sistemaro runboat testala pure e se ok approva cosi c'e' possibilita' di merge

Ciao,
quindi da dove scaricare la versione modificata? Scusa per l'ignoranza ma non sono molto avvezzo con github

**Comment by matteoopenf** [comment]
Ciao,Ho postato la pr collegata nella issue.Una volta che viene approvata mergiataTi basterà un git pull sul repo che hai in locale di l10n-italyMatteo ***@***.*** Il giorno 9 mag 2024, alle ore 10:41, pezcl67 ***@***.***> ha scritto:﻿


Ciao, ho proprio questo tipo di problema con la epson. La modifica fatta è testabile? Verrà messa nel branch a breve?
Grazie mille

ho sistemaro runboat testala pure e se ok approva cosi c'e' possibilita' di merge

Ciao,
quindi da dove scaricare la versione modificata? Scusa per l'ignoranza ma non sono molto avvezzo con github

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

Openforce Srls Unipersonale | www.openforce.it | 02 21119472 | Via Tiziano,32 20145 Milano

**Comment by matteoopenf** [comment]
Oppure  a mano inserisci il codice della pr in locale https://github.com/OCA/l10n-italy/pull/3722  e vedi se tutto funziona e poi approvi la PR

**Comment by pezcl67** [comment]
Ciao,
ho installato la modifica e provato se funziona correttamente.
Purtroppo la modifica si è rivelata peggiorativa.
Quasi il 30% degli scontrini andavano in errore e la stampante mandava un annullo.
Ti allego due esempi dei tanti scontrini annullati.

A volte sul display della Epson (nei casi di errore) dava un resto di 1 centesimo (con scritta "Differenza 0,01€).
[20240514095325311.pdf](https://github.com/OCA/l10n-italy/files/15304892/20240514095325311.pdf)

**Comment by matteoopenf** [comment]
> Ciao, ho installato la modifica e provato se funziona correttamente. Purtroppo la modifica si è rivelata peggiorativa. Quasi il 30% degli scontrini andavano in errore e la stampante mandava un annullo. Ti allego due esempi dei tanti scontrini annullati.
> 
> A volte sul display della Epson (nei casi di errore) dava un resto di 1 centesimo (con scritta "Differenza 0,01€). [20240514095325311.pdf](https://github.com/OCA/l10n-italy/files/15304892/20240514095325311.pdf)

Grazie Mille del Feedback, appena posso contatto il cliente a cui ho fornito questo fix per capire se hanno problemi e non ci hanno detto nulla, perche' comunque sono circa 6 mesi che glielo abbiamo dato non 1 giorno.
In ogni caso grazie!

**Comment by pezcl67** [comment]
Grazie a te.
Mi potresti dare un contatto mail? Volevo qualche informazione sulla tua azienda per eventuali sviluppi.
Grazie

**Comment by matteoopenf** [comment]
> Grazie a te. Mi potresti dare un contatto mail? Volevo qualche informazione sulla tua azienda per eventuali sviluppi. Grazie

info@openforce.it scrivici pure qua e poi un collega ti ricontattano i colleghi

**Comment by pezcl67** [comment]
Ok, grazie.
Fai anche sviluppo come libero professionista ?

**Comment by matteoopenf** [comment]
> Ok, grazie. Fai anche sviluppo come libero professionista ?

No, sono dipendente per qualsiasi cosa fai riferimento alla mail sopra.
